src/main/scala/scalismo/color/RGBA.scala

```diff
@@ -0,0 +1,211 @@
+  /**
+   * convert to AWT default color
+   * expects a clamped color value
+   */
+  def toAWTColor: Color = new java.awt.Color(r.toFloat, g.toFloat, b.toFloat, a.toFloat)
+}
+
+object RGBA {
+
+  val White: RGBA = RGBA(1.0, 1.0, 1.0, 1.0)
+  val Black: RGBA = RGBA(0.0, 0.0, 0.0, 1.0)
+
+  val WhiteTransparent: RGBA = RGBA(1.0, 1.0, 1.0, 0.0)
+  val BlackTransparent: RGBA = RGBA(0.0, 0.0, 0.0, 0.0)
+
+  def apply(color: RGB): RGBA = new RGBA(color.r, color.g, color.b, 1.0)
+  def apply(color: RGB, a: Double): RGBA = new RGBA(color.r, color.g, color.b, a)
+  def apply(r: Double, g: Double, b: Double): RGBA = new RGBA(r, g, b, 1.0)
+  def apply(gray: Double): RGBA = new RGBA(gray, gray, gray, 1.0)
+  def apply(gray: Double, a: Double): RGBA = new RGBA(gray, gray, gray, a)
+  def apply(tuple: (Double, Double, Double, Double)) = new RGBA(tuple._1, tuple._2, tuple._3, tuple._4)
+  def apply(awtColor: Color): RGBA = RGBA(fromInt8(awtColor.getRed), fromInt8(awtColor.getGreen), fromInt8(awtColor.getBlue), fromInt8(awtColor.getAlpha))
+
+  /** implementation of the Vectorizer interface for RGBA */
+  implicit object RGBAComponents extends ComponentRepresentation[RGBA] with Vectorizer[RGBA] {
+    override def fromArray(arr: Array[Double]): RGBA = {
+      require(arr.length == size)
+      RGBA(arr(0), arr(1), arr(2), arr(3))
+    }
+
+    override def toArray(color: RGBA): Array[Double] = Array(color.r, color.g, color.b, color.a)
+
+    override def intoArray(color: RGBA, array: Array[Double]): Array[Double] = {
+      require(array.length == 4)
+      array(0) = color.r
+      array(1) = color.g
+      array(2) = color.b
+      array(3) = color.a
+      array
+    }
+
+    override val size: Int = 4
+
+    override def component(color: RGBA, index: Int): Double = (index: @switch) match {
+      case 0 => color.r
+      case 1 => color.g
+      case 2 => color.b
+      case 3 => color.a
+      case _ => throw new Exception(s"invalid index ($index) in RGBA vectorizer")
+    }
+
+    override def fromComponents(comp: (Int) => Double): RGBA = RGBA(comp(0), comp(1), comp(2), comp(3))
+
+    override def dim: Int = 4
+
+    override def vectorize(v: RGBA): DenseVector[Double] = DenseVector(v.r, v.g, v.b, v.a)
+
+    override def unvectorize(d: DenseVector[Double]): RGBA = RGBA(d(0), d(1), d(2), d(3))
+  }
+
+  implicit object RGBAOperations extends ColorSpaceOperations[RGBA] {
+    /** add two pixels */
+    override def add(pix1: RGBA, pix2: RGBA): RGBA = pix1 + pix2
+
+    /** scalar multiplication */
+    override def scale(pix: RGBA, l: Double): RGBA = pix * l
+
+    /** dot product */
+    override def dot(pix1: RGBA, pix2: RGBA): Double = pix1 dot pix2
+
+    /** channel-wise multiplication */
+    override def multiply(pix1: RGBA, pix2: RGBA): RGBA = pix1 x pix2
+
+    /** zero element */
+    override def zero: RGBA = BlackTransparent
+
+    override val dimensionality = 4
+  }
+
+  implicit object RGBAInterpolator extends ValueInterpolator[RGBA] {
```

Comment:
Am sure there is a viable explanation, but it is not clear to me why the interpolator (and consequently color space operations) would also have to be ported?